app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt

```diff
@@ -177,23 +177,49 @@ class UserRestoreInteractor(
     ): RestoreUserResult {
         var result: RestoreUserResult
 
+        // First case try to restore with DEVICE + SOCIAL
         result = tryRestoreUser(OnboardingFlow.RestoreWallet.DevicePlusSocialShare)
 
         if (result is RestoreUserResult.RestoreFailure) {
+            // if restore was failed
+            // Try last try with SOCIAL + CUSTOM
             result = tryRestoreUser(OnboardingFlow.RestoreWallet.SocialPlusCustomShare)
         }
-        return result
+
+        // if restore is FAILED we throw error thast SHARES ARE NOT MATCH
+        // otherwise Restore is SUCCESSFULL
+        return when (result) {
+            is RestoreUserResult.RestoreFailure -> {
+                RestoreUserResult.RestoreFailure.DevicePlusSocialOrSocialPlusCustom(
+                    RestoreUserException("Shares are not match")
```

Comment:
Shares device+social or social+custom are not match

Reply:
same with all pieces of code

Reply:
Yes handling the same, i think it's not so bad, we can change it after. Now our target to pass this feature , ASAP))